File: server/parse_static_blogs.py
import re
import json

def parse_blogs():
    with open('staticBlogs.ts', 'r', encoding='utf-8') as f:
        content = f.read()
    
    # We want to extract the JSON objects in the STATIC_BLOGS array.
    # A simple regex to find the blocks starting with { and having id: or "id":
    # Let's find all occurrences of "id": or id: in the file.
    matches = re.finditer(r'(id|"\bid\b")\s*:\s*(\d+)', content)
    
    blogs = []
    for match in matches:
        start_idx = match.start()
        # Find the title, slug, and image_url around this start_idx
        # Let's search forward from start_idx for title, slug, image_url
        substring = content[start_idx:start_idx + 1000]
        
        id_val = match.group(2)
        
        title_match = re.search(r'"title"\s*:\s*"([^"]+)"', substring)
        if not title_match:
            title_match = re.search(r'title\s*:\s*"([^"]+)"', substring)
            
        slug_match = re.search(r'"slug"\s*:\s*"([^"]+)"', substring)
        if not slug_match:
            slug_match = re.search(r'slug\s*:\s*"([^"]+)"', substring)
            
        img_match = re.search(r'"image_url"\s*:\s*"([^"]+)"', substring)
        if not img_match:
            img_match = re.search(r'image_url\s*:\s*"([^"]+)"', substring)
            
        title = title_match.group(1) if title_match else "Unknown"
        slug = slug_match.group(1) if slug_match else "Unknown"
        img = img_match.group(1) if img_match else "Unknown"
        
        blogs.append({
            "id": int(id_val),
            "title": title,
            "slug": slug,
            "image_url": img
        })
        
    print(json.dumps(blogs, indent=2))

if __name__ == '__main__':
    parse_blogs()

File: server/parse_static_blogs.js
const fs = require('fs');
const path = require('path');

function parseBlogs() {
    const filePath = path.join(__dirname, '../staticBlogs.ts');
    const content = fs.readFileSync(filePath, 'utf8');

    // Regex to match the structures
    // Let's search for id: and fields
    const idRegex = /"id"\s*:\s*(\d+)|id\s*:\s*(\d+)/g;
    const blogs = [];
    let match;

    while ((match = idRegex.exec(content)) !== null) {
        const idVal = match[1] || match[2];
        const startIdx = match.index;
        const substring = content.substring(startIdx, startIdx + 1000);

        const titleMatch = substring.match(/"title"\s*:\s*"([^"]+)"/) || substring.match(/title\s*:\s*"([^"]+)"/);
        const slugMatch = substring.match(/"slug"\s*:\s*"([^"]+)"/) || substring.match(/slug\s*:\s*"([^"]+)"/);
        const imgMatch = substring.match(/"image_url"\s*:\s*"([^"]+)"/) || substring.match(/image_url\s*:\s*"([^"]+)"/);

        blogs.append = blogs.push({
            id: parseInt(idVal),
            title: titleMatch ? titleMatch[1] : 'Unknown',
            slug: slugMatch ? slugMatch[1] : 'Unknown',
            image_url: imgMatch ? imgMatch[1] : 'Unknown'
        });
    }

    console.log(JSON.stringify(blogs, null, 2));
}

parseBlogs();
